tensor_test.go

```diff
@@ -123,3 +125,21 @@ func TestTensorPinMemory(t *testing.T) {
 		assert.Equal(t, " 1  2\n 3  4\n[ CPUFloatType{2,2} ]", b.String())
 	}
 }
+
+func TestTensorGC(t *testing.T) {
+	torch.GC()
+	runtime.LockOSThread()
+	c := make(chan bool, 0)
+	{
+		torch.NewTensor([][]float32{{1, 2}, {3, 4}})
+		go func() {
+			a := torch.NewTensor([][]float32{{1, 2}, {3, 4}})
+			c <- true
+			time.Sleep(time.Second)
+			runtime.KeepAlive(&a)
```

Comment:
Is the purpose of calling `KeepAlive` to check that `torch.GC` will free up two tensors -- on created in L134 and the one in L136?  However, I  don't see a checking in this test case for the freeing up of two tensors?

Reply:
This is to tell the escape analysis of Go garbage collector not to mark `a` as unreachable before the call of `KeepAlive`. 

Without the `thread_local`, `torch.GC()` will recycle the `Tensor` in L134 and hang until the goroutine ends, and then recycle the `Tensor` in L136. 

In practice, without this fix, if we keep a cache and fill it in a goroutine, calling `torch.GC` in the main goroutine will certainly hang the training process forever.